pkg/service/v1alpha1/controller.go

```diff
@@ -0,0 +1,376 @@
+/*
+Copyright © 2018-2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	utils "github.com/openebs/zfs-localpv/pkg/agent/v1alpha1"
+	errors "github.com/openebs/zfs-localpv/pkg/common/errors/v1alpha1"
+	csipayload "github.com/openebs/zfs-localpv/pkg/payload/v1alpha1"
+	csivol "github.com/openebs/zfs-localpv/pkg/volume/v1alpha1"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// controller is the server implementation
+// for CSI Controller
+type controller struct {
+	driver       *CSIDriver
```

Comment:
what is the role of driver. I dont find it being used.

Reply:
can we add volume properties into this structure and eventually get rid of csi related imports from this package